src/imagery/i.eodag/i.eodag.py

```diff
@@ -0,0 +1,319 @@
+        return 0
+
+
+def get_bb(vector=None):
+    args = {}
+    if vector:
+        args["vector"] = vector
+    # are we in LatLong location?
+    kv = gs.parse_command("g.proj", flags="j")
+    if "+proj" not in kv:
+        gs.fatal("Unable to get bounding box: unprojected location not supported")
+    if kv["+proj"] != "longlat":
+        info = gs.parse_command("g.region", flags="uplg", **args)
+        return {
+            "lonmin": info["nw_long"],
+            "latmin": info["sw_lat"],
+            "lonmax": info["ne_long"],
+            "latmax": info["nw_lat"],
+        }
+    else:
+        info = gs.parse_command("g.region", flags="upg", **args)
+        return {
+            "lonmin": info["w"],
+            "latmin": info["s"],
+            "lonmax": info["e"],
+            "latmax": info["n"],
+        }
+
+
+def download_products(products_list):
+    dag.download_all(products_list)
+
+
+def download_by_id(query_id: str):
+    print(f"Searching for product ending with: {query_id[-min(len(query_id), 8):]}")
+    product, count = dag.search(id=query_id)
+    if count != 1:
+        raise ParameterError("Product couldn't be uniquely identified")
+    if not product[0].properties["id"].startswith(query_id):
+        raise ParameterError("Product wasn't found")
+    print(f"Poduct ending with: {query_id[-min(len(query_id), 8):]} is found.")
+    print("Downloading...")
+    dag.download(product[0])
+
+
+def download_by_ids(ids: list):
+    for id in ids:
+        try:
+            download_by_id(id)
+        except ParameterError:
+            gs.error(
+                f"Product ending with: {id[-min(len(id), 8):]}, failed to download"
+            )
+
+
+def setup_environment_variables():
+    os.environ[f"EODAG__{options['provider']}__DOWNLOAD__EXTRACT"] = str(flags["e"])
+    os.environ[f"EODAG__{options['provider']}__DOWNLOAD__DELETE_ARCHIV"] = str(
+        flags["d"]
+    )
+
+    if options["output"]:
+        os.environ[f"EODAG__{options['provider']}__DOWNLOAD__OUTPUTS_PREFIX"] = options[
+            "output"
+        ]
+
+
+def main():
+    # products: https://github.com/CS-SI/eodag/blob/develop/eodag/resources/product_types.yml
+
+    # setting the envirionmnets variable has to come before the dag initialization
+    setup_environment_variables()
+
+    global dag
+    dag = EODataAccessGateway()
+
+    if options["provider"]:
+        dag.set_preferred_provider(options["provider"])
+
+    # Download by ids... if ids are provided only these ids will be downloaded
+    if options["id"]:
+        ids = options["id"].split(",")
+        download_by_ids(ids)
+    else:
+        # TODO: Add option for setting a differnt config file path
+
+        items_per_page = 20
+        # TODO: Check that the product exists, could be handled by catching exceptions when searching...
+        product_type = options["dataset"]
+
+        # TODO: Allow user to specify a shape file path
```

Comment:
If one wors in projected CRS it may be more convenient to be able to provide tha AOI in e.g. a GeoJSON... In principle any OGR readable format could be OK...

Reply:
Handled further here: https://github.com/HamedElgizery/grass-addons/issues/3